src/BlockTemplatesController.php

```diff
@@ -51,6 +51,7 @@ protected function init() {
 		add_action( 'template_redirect', array( $this, 'render_block_template' ) );
 		add_filter( 'pre_get_block_file_template', array( $this, 'get_block_file_template' ), 10, 3 );
 		add_filter( 'get_block_templates', array( $this, 'add_block_templates' ), 10, 3 );
+		add_filter( 'search_template_hierarchy', array( $this, 'update_search_template_hierarchy' ), 10, 3 );
```

Comment:
I'm not sure that we should put the logic for the Product Search Result page here. I would like to have a separation of concerns here. The Product Search Results is a custom template that has custom logic

Reply:
I agree with @gigitux on this comment.